### PR TITLE
Advanced modal dialog

### PR DESCRIPTION
**Pull request by sihil**
This radically improves the UI for doing cross stack restores.

The new dialog for restoring from one stack to another looks like this:
screen shot 2016-07-01 at 19 40 31

Thanks to @alastairjardine and @ShaunYearStrong for their help with this.

### CODE REVIEW

**Comment by sihil** [comment]
Be great if you could have a look a this @clloyd - in particular the CSS works but might not be done optimally, or named according to the existing style. Suggestions either way would be greatly appreciated.

**Comment by alastairjardine** [comment]
Looks great @sihil. Assuming that you've run it past some target users, gets a 👍 from me!

**Comment by alastairjardine** [comment]
(Oops minor comment: drop the "of" from the date e.g. "4th of July" to "4th July")

File: public/sass/components/modal.scss [hunk 1]
```diff
@@ -57,6 +58,74 @@ gu-box.modal__content {
   margin-left: 10px;
 }
 
+.modal__content__destination__list {
```
Comment (clloyd) [inline public/sass/components/modal.scss]: The naming convention is BEM here, so each __ means the subcomponent. 

So if there's a `destination` component with a  `list` component within then that name is correct, but if there is a `destination list` component then it's clearer to use `.modal__content__destination-list`

**Comment by sihil** [comment]
@alastairjardine I've extracted out the date formatting to make it consistent, including the superscript ordinal

**Pull request merged**